src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -366,9 +374,36 @@ public static RowIterator cas(String keyspaceName,
         }
 
         ConsensusAttemptResult lastAttemptResult;
+        Epoch lastEpoch = null;
         do
         {
+            if (lastEpoch != null)
+            {
+
+                long timeout = requestTime.computeTimeout(nanoTime(), DatabaseDescriptor.getTransactionTimeout(NANOSECONDS));
+                Epoch lastEpochFinal = lastEpoch;
+                ClusterMetadataService.instance().log().highestPending().ifPresent(epoch ->
+                       {
+                           try
+                           {
+                               ClusterMetadataService.instance().awaitAtLeast(epoch, timeout, NANOSECONDS);
+                           }
+                           catch (InterruptedException e)
+                           {
+                               throw new RuntimeException(e);
```

Comment:
```suggestion
                               Thread.currentThread().interrupt();
                               throw new UncheckedInterruptedException(e);
```

Reply:
We shouldn't need to invoke Thread.interrupt(), but good catch point out the change to `UncheckedInterruptedException`